src/libs/common/types/Location.ts

```diff
@@ -0,0 +1,33 @@
+import type { URI } from "vscode-uri";
+import type { Position } from "@cursorless/common";
+import { Range } from "@cursorless/common";
+
+/**
+ * Represents a location inside a resource, such as a line
+ * inside a text file.
+ */
+export class Location {
```

Comment:
Hmm.  I thought we decided we didn't want to mirror their `Location` interface.  Do we use this in a lot of places?  Might be nice if we could avoid it for now

Reply:
We don't but for now we are required because it's used by `getNodeAtLocation`. I was thinking of moving that from the graph and into the document instead in a follow up pr.

Reply:
Okay I went a here then actually moved it. The nice thing about this change is that our scope handlers now have access to `getNodeAtLocation`

Reply:
Hmm.  I thought about it more, and I don't think we want the whole world having access to the parse tree, as it is pretty volatile and low-level.  How hard is it to roll back that change?

Reply:
Sounds good

Reply:
Update: we hesitantly decided to roll it back, and to keep an eye on how much more painful all of this "good engineering practice" makes it to change our code base